packages/wonder-blocks-birthday-picker/src/components/birthday-picker.js

```diff
@@ -0,0 +1,259 @@
+     * format or `null`.
+     */
+    onChange: (date: ?string) => mixed,
+|};
+
+type State = {|
+    /**
+     * The currently selected month.
+     */
+    month: string | null,
+    /**
+     * The currently selected day.
+     */
+    day: string | null,
+    /**
+     * The currently selected year.
+     */
+    year: string | null,
+    /**
+     * The error message to display (in case there's an invalid date).
+     */
+    error: string | null,
+|};
+
+// Flow doesn't know about the getYear property on Date for some reason!
+// $FlowFixMe[prop-missing]
+const CUR_YEAR = new Date().getYear() + 1900;
+// TODO(WB-1200.2): Add labels prop to handle translations.
+const BIRTHDAY_ERROR = "Please select a valid birthdate.";
+
+/**
+ * Birthday Picker. Similar to a datepicker, but specifically for birthdates.
+ * We don't want to show a calendar in this case as it can be quite tedious to
+ * try and select a date that's many years old. Instead, we use a set of
+ * dropdowns to achieve a similar effect.
+ *
+ * More information on this pattern:
+ * https://medium.com/samsung-internet-dev/making-input-type-date-complicated-a544fd27c45a
+ *
+ * Arguably, this should probably even be 3 textfields, but that would be a
+ * larger design change, more info:
+ * https://designnotes.blog.gov.uk/2013/12/05/asking-for-a-date-of-birth/
+ *
+ * **NOTE:** This component is uncontrolled.
+ *
+ * ### Usage
+ *
+ * ```jsx
+ * import {BirthdayPicker} from "@khanacademy/wonder-blocks-dates";
+ *
+ * <BirthdayPicker
+ *  defaultValue="2021-05-26"
+ *  onChange={(date) => {setDate(date)}}
+ * />
+ * ```
+ */
+export default class BirthdayPicker extends React.Component<Props, State> {
+    constructor(props: Props) {
+        super(props);
+
+        this.lastChangeValue = props.defaultValue || null;
+        this.state = this.getStateFromDefault();
+    }
+
+    /**
+     * Calculates the initial state values based on the default value.
+     */
+    getStateFromDefault(): State {
+        const {defaultValue} = this.props;
+        const initialState: State = {
+            month: null,
+            day: null,
+            year: null,
+            error: null,
+        };
+
+        // If a default value was provided then we use moment to convert it
+        // into a date that we can use to populate the
+        if (defaultValue) {
+            const date = moment(defaultValue);
+
+            if (date.isValid()) {
+                initialState.month = String(date.month());
+                initialState.day = String(date.date());
+                initialState.year = String(date.year());
+            }
+
+            // If the date is in the future or is invalid then we want to show
+            // an error to the user.
+            if (date.isAfter() || !date.isValid()) {
+                initialState.error = BIRTHDAY_ERROR;
+            }
+        }
+
+        return initialState;
+    }
```

Comment:
Nice. Could completely decouple this from the component by taking the default in as an arg - no change requested.